version: '3.8'

services:
    web:
        restart: always
        networks:
            - external_dev_vlan20
            - net_192_168_y
        dns: 192.168.x.1
        expose:
            - "80"
        volumes:
            - ./site.conf:/etc/nginx/conf.d/default.conf
        image: nginx:latest
        labels:
            - "traefik.docker.network=externaldev_vlan20"
            - "traefik.enable=true"
            - "traefik.http.routers.web-http2.rule=Host(`gelukkig2021.nl`) || Host(`www.gelukkig2021.nl`)"
            - "traefik.http.routers.web-http2.entrypoints=web"
            - "traefik.http.services.web2.loadbalancer.server.port=80"
            - "traefik.http.routers.web2.rule=Host(`gelukkig2021.nl`) || Host(`www.gelukkig2021.nl`)"
            - "traefik.http.routers.web2.entrypoints=websecure"
            - "traefik.http.routers.web2.tls=true"
            - "traefik.http.routers.web2.tls.certresolver=leresolver"
            - "traefik.http.routers.web-http2.middlewares=redirect-https"
            - "traefik.http.middlewares.redirect-https.redirectscheme.scheme=https"
            - "traefik.http.middlewares.redirect-https.redirectscheme.port=443"
            - "traefik.docker.network=external_dev_vlan20"
        logging:
            driver: local
            options:
                max-size: "100m"
                max-file: "3"

networks:
    external_dev_vlan20:
        external: true
    net_192_168_y:
        external: true
